pkg/webservice/handlers.go

```diff
@@ -534,15 +534,18 @@ func getQueueApplications(w http.ResponseWriter, r *http.Request) {
 		buildJSONErrorResponse(w, QueueDoesNotExists, http.StatusBadRequest)
 		return
 	}
-	apps := queue.GetCopyOfApps()
-	completedApps := queue.GetCopyOfCompletedApps()
-	appsDao := make([]*dao.ApplicationDAOInfo, 0, len(apps)+len(completedApps))
-	for _, app := range apps {
+
+	appsDao := make([]*dao.ApplicationDAOInfo, 0)
+	for _, app := range queue.GetCopyOfApps() {
 		appsDao = append(appsDao, getApplicationJSON(app))
 	}
-	for _, app := range completedApps {
-		appsDao = append(appsDao, getApplicationJSON(app))
+
+	for _, app := range partitionContext.GetCompletedApplications() {
+		if app.GetQueuePath() == queueName {
+			appsDao = append(appsDao, getApplicationJSON(app))
+		}
 	}
+
```

Comment:
I would suggest that we remove that and add a release note for the same to YuniKorn 1.2 explaining the reasoning behind the change.

Reply:
Agreed. We can deprecate the field in the 1.2 docs, and remove in 2.0. It will just be empty until then.

Reply:
SGTM. Addressed in latest commits.